src/main/java/com/risevision/bigquery/BigqueryJobsApi.java

```diff
@@ -0,0 +1,63 @@
+package com.risevision.bigquery;
+
+import com.google.api.client.http.HttpTransport;
+import com.google.api.services.bigquery.model.*;
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Logger;
+
+public class BigqueryJobsApi extends BigqueryCommonApi implements JobsApi{
+
+  BigqueryJobsApi(HttpTransport transport) {
+    super(transport);
+  }
+
+  static final Logger log = Logger.getAnonymousLogger();
+
+  @Override
+  public void insertJob(JobInserterConfig configuration,
+                        JobInserterConfig.IndividualQueryConfiguration queryConfiguration) {
+    try {
+      Job job = assembleJob(configuration, queryConfiguration);
+      bqClient.jobs().insert(configuration.projectId, job).execute();
+    } catch (IOException e) {
+      log.severe("Could not insert job " + e.getMessage());
+    }
+  }
+
+  protected Job assembleJob(JobInserterConfig configuration,
+                  JobInserterConfig.IndividualQueryConfiguration queryConfiguration) throws IOException {
+
+    Job job = new Job();
+    JobConfiguration jobConfiguration = new JobConfiguration();
+    JobConfigurationQuery jobConfigurationQuery = new JobConfigurationQuery();
+    JobReference jobReference = new JobReference();
+    String sqlQuery = readResource(queryConfiguration.fileName);
+    TableReference destinationTable = new TableReference();
+
+    // Prepare BigQuery request objects
+    job.setJobReference(jobReference);
+    job.setConfiguration(jobConfiguration);
+    jobConfiguration.setQuery(jobConfigurationQuery);
+    jobConfigurationQuery.setDestinationTable(destinationTable);
+
+    // Configuration common to all jobs
+    jobReference.setProjectId(configuration.projectId);
+    jobConfigurationQuery.setWriteDisposition(configuration.configuration.query.writeDisposition);
+    destinationTable.setProjectId(configuration.configuration.query.destinationTable.projectId);
+    destinationTable.setDatasetId(configuration.configuration.query.destinationTable.datasetId);
+
+    // Job-specific configuration
+    jobConfigurationQuery.setQuery(sqlQuery);
+    destinationTable.setTableId(queryConfiguration.destinationTableId);
+
+    return job;
+  }
+
+  String readResource(String fileName) throws IOException {
```

Comment:
@tejohnso one concern I have is that `readResource` does not attempt to handle exceptions, and the layers that call it don't either. This could cause an entire batch of jobs to fail (if, say, a query file is missing or there's a typo)

I chose not to handle them because it's reasonable to crash on configuration errors, and we have a PR review process to catch this sort of mistake, but I'm wondering what your opinion is.